_pages/fsa/features/fsa-checkout.md

```diff
@@ -0,0 +1,116 @@
+The following example of payment checkout step definition restricts payment step for all banking products:
+```
+     {
+        id: 'checkoutPaymentDetailsStep',
+        name: 'fscommon.paymentDetails',
+        routeName: 'checkoutPaymentDetails',
+        type: [CheckoutStepType.PAYMENT_DETAILS],
+        restrictedCategories: [
+          'banking_credit_card',
+          'banking_current_account',
+          'banking_loans',
+          'banking_fixed_term_deposit',
+        ],
+      },
+```
+
+## Components
+
+FSCheckoutModule consists of multiple checkout components, guards and services, making the checkout process fully configurable. 
+The following example shows step components inside the FSCheckoutModule:
+```
+    AddOptionsComponent,
+    QuoteReviewComponent,
+    BindQuoteDialogComponent,
+    ReferredQuoteDialogComponent,
+    FinalReviewComponent,
+    ChooseCoverNavigationComponent,
+    PersonalDetailsNavigationComponent,
+    OrderConfirmationComponent,
+    OrderConfirmationMessageComponent,
+    MiniCartComponent,
+```
+
+These components are fulfilled with logic from *FSCheckoutConfigService*, responsible for navigation by setting next and previous steps. 
+If you want to extend your checkout with an additional step, you should add the step definition and route to *default-checkout-config.ts*, create a new component and place the logic for handling steps inside it.
+
+QuoteReviewComponent example of implementing nextCheckoutStep:  
+```
+@Component({
+    selector: 'cx-fs-quote-review',
+    templateUrl: './quote-review.component.html',
+  })
+ export class QuoteReviewComponent implements OnInit, OnDestroy {`
+ constructor(
+    protected cartService: FSCartService,
+    protected config: OccConfig,
+    protected routingService: RoutingService,
+    protected checkoutConfigService: FSCheckoutConfigService,
+    protected categoryService: CategoryService,
+    protected activatedRoute: ActivatedRoute,
+    protected modalService: ModalService,
+    protected translationService: FSTranslationService
+  ) {}
+
+  ngOnInit() {
+    this.cart$ = this.cartService.getActive();
+    this.isCartStable$ = this.cartService.isStable();
+    this.previousCheckoutStep$ = this.checkoutConfigService.previousStep;
+    this.nextCheckoutStep$ = this.checkoutConfigService.nextStep;
+    this.activeCategory$ = this.categoryService.getActiveCategory();
+  }
+```
+
+quote-review.component.html
+
+```html
+  <ng-container
+                *ngIf="nextCheckoutStep$ | async as nextCheckoutStep"
+                ><button
+                  class="primary-button btn-block"
+                  type="button"
+                  data-checkout-url=""
+                  (click)="navigateNext(nextCheckoutStep, cart)"
+                >
+                  {{ 'common.continue' | cxTranslate }}
+                </button></ng-container
+              >
+```
+
+Many guards have been implemented to prevent access to restricted or uncompleted checkout steps:
```

Comment:
@divna-p If these are all the guards that have been implemented, we could then say: 

The following guards have been implemented to prevent access to restricted or uncompleted checkout steps:

Reply:
Yes, fixed.